rpcs3/rpcs3qt/debugger_list.cpp

```diff
@@ -139,21 +170,69 @@ void debugger_list::keyPressEvent(QKeyEvent* event)
 
 	switch (event->key())
 	{
-	case Qt::Key_PageUp:   ShowAddress(m_pc - (m_item_count * 4), true); return;
-	case Qt::Key_PageDown: ShowAddress(m_pc + (m_item_count * 4), true); return;
-	case Qt::Key_Up:       ShowAddress(m_pc - 4, true); return;
-	case Qt::Key_Down:     ShowAddress(m_pc + 4, true); return;
+	case Qt::Key_PageUp:   scroll(0 - m_item_count); return;
+	case Qt::Key_PageDown: scroll(m_item_count); return;
+	case Qt::Key_Up:       scroll(1); return;
+	case Qt::Key_Down:     scroll(-1); return;
 	default: break;
 	}
 }
 
+void debugger_list::create_rsx_command_detail(u32 pc, int row)
+{
+	RSXDisAsm rsx_dis = CPUDisAsm::copy_and_change_mode(*static_cast<RSXDisAsm*>(m_disasm), CPUDisAsmMode::List);
+
+	for (; row > 0; row--)
+	{
+		// Skip methods
+		pc += std::max<u32>(rsx_dis.disasm(pc), 4);
+	}
+
+	// Either invalid or not a method
+	if (rsx_dis.disasm(pc) <= 4) return;
+
+	if (m_cmd_detail)
+	{
+		// Edit the existing dialog
+		m_detail_label->setText(QString::fromStdString(rsx_dis.last_opcode));
+		m_cmd_detail->setFixedSize(m_cmd_detail->sizeHint());
+		m_cmd_detail->move(QCursor::pos());
+		return;
+	}
+
+	m_cmd_detail = new QDialog(this);
+	m_cmd_detail->setWindowTitle(tr("RSX Command Detail"));
+
+	m_detail_label = new QLabel(QString::fromStdString(rsx_dis.last_opcode), this);
+	m_detail_label->setFont([](QFont f) { f.setPointSize(9); return f; }(m_detail_label->font()));
```

Comment:
Maybe create a gui util function for the font. I'Ve seen you duplicating this code

And maybe consider mono font here

Reply:
Realized I can copy the font of debugger_list.